pkg/controller/dataimportcron-controller.go

```diff
@@ -704,24 +941,48 @@ func addDataImportCronControllerWatches(mgr manager.Manager, c controller.Contro
 	getCronName := func(obj client.Object) string {
 		return obj.GetLabels()[common.DataImportCronLabel]
 	}
-	mapToCron := func(obj client.Object) []reconcile.Request {
+	mapSourceObjectToCron := func(obj client.Object) []reconcile.Request {
 		if cronName := getCronName(obj); cronName != "" {
 			return []reconcile.Request{{NamespacedName: types.NamespacedName{Name: cronName, Namespace: obj.GetNamespace()}}}
 		}
 		return nil
 	}
+	mapStorageProfileToCron := func(obj client.Object) (reqs []reconcile.Request) {
+		var crons cdiv1.DataImportCronList
+		if err := mgr.GetClient().List(context.TODO(), &crons); err != nil {
+			c.GetLogger().Error(err, "Unable to list DataImportCrons")
+			return
+		}
+		// Storage profiles are 1:1 to storage classes
+		scName := obj.GetName()
+		for _, cron := range crons.Items {
+			dataVolume := cron.Spec.Template
+			explicitScName := getStorageClassFromTemplate(&dataVolume)
+			templateSc, err := cc.GetStorageClassByName(mgr.GetClient(), explicitScName)
+			if err != nil || templateSc == nil {
+				c.GetLogger().Error(err, "Unable to get storage class", "templateSc", templateSc)
+				return
+			}
+			if templateSc.Name == scName {
```

Comment:
why not annotate the DIC with templateSc.Name in the reconcile (you are already getting it there), so you use an index to filter the List() and get rid of the loop.

Reply:
I can do that, but, the storage profile watch going off should be a rare occurrence, may not be worth the index?
not sure what's the overhead other than managing an annotation

Reply:
Afaiu indexing won't cost too much as their values will be static and there won't too many DICs.

Reply:
done

Reply:
Reverted the index with a TODO;
As we offline discussed, we may lose the storage profile event, as relying on the storage class annotation may get racy:
Even if we set up the indexer before the controller, we would still not have the annotation yet.

Kept the annotation so we could capitalize on it later when we switch to indexing here

Reply:
You may append in `reqs` the crons with the expected scName annotation, plus the crons without the scName annotation yet, and it solves the race afaiu.

Reply:
we can and it does, but I don't think it's much better than the current approach
a little bit more inversed logically, still no index, still List on all of them

Reply:
Ok, not sure it's the best choice, but I can accept your argument...